### DISCUSSION TITLE
Graclus Issue

### DISCUSSION THREAD

**Discussion by yikang7720** [pyg-team/pytorch_geometric#4621] [Q&A]
Hi,

I find a bug exists when I am using the graclus function with weight. This is the code of the graclus network for graph classification. However, I find that if I set the weight of the graclus function as my data weight. The network will run all the time and no results or accuracy provided. 

Here is the my Data.
Data(edge_index=[2, 13456], weight=[13456], num_nodes=116, x=[116, 1], y=0).

Here is the code for the graclus network.
`import torch
import torch.nn.functional as F
from torch.nn import Linear

from torch_geometric.data import Batch
from torch_geometric.nn import (
    GraphConv,
    JumpingKnowledge,
    global_mean_pool,
    graclus,
    max_pool,
)

num_layers  = 4
hidden = 16
class Graclus(torch.nn.Module):
    def __init__(self):
        super(Graclus, self).__init__()
        self.conv1 = GraphConv(1, 16, aggr='mean')
        self.convs = torch.nn.ModuleList()
        for i in range(num_layers - 1):
            self.convs.append(GraphConv(16, 16, aggr='mean'))
        self.jump = JumpingKnowledge(mode='cat')
        self.lin1 = Linear(64, 16)
        self.lin2 = Linear(16, 1)

    def forward(self, data):
        x, edge_index, weight , batch = data.x, data.edge_index, data.weight, data.batch
        x = F.relu(self.conv1(x, edge_index))
        xs = [global_mean_pool(x, batch)]
        for i, conv in enumerate(self.convs):
            x = F.relu(conv(x, edge_index))
            xs += [global_mean_pool(x, batch)]
            if i % 2 == 0 and i < len(self.convs) - 1:
                cluster = graclus(edge_index,num_nodes=x.size(0))
                data = Batch(x=x, edge_index=edge_index, batch=batch)
                data = max_pool(cluster, data)
                x, edge_index, batch = data.x, data.edge_index, data.batch
        x = self.jump(xs)
        x = F.relu(self.lin1(x))
        x = self.lin2(x)
        return x`

**Comment by rusty1s** [comment]
Can you share your `edge_index` representation with us? How does `data.weight` look like? Does it work when enforcing that its values are inside `[0, 1)`?

**Reply by yikang7720** [reply]
The edge_index is the connections between different nodes.
It looks like the 
'tensor([[  0,   0,   0,  ..., 115, 115, 115],
                                       [  0,   1,   2,  ..., 113, 114, 115]]).'

The edge weight looks like this 'tensor([1.0000, 0.7060, 0.5595,  ..., 0.0961, 0.6627, 1.0000])'

I think the value of data.weight is already around [0,1).

**Reply by rusty1s** [reply]
Can you upload the data somewhere generated via `torch.save((edge_index, edge_weight), 'data.pt')`?
